src/ShadowDOM/wrappers/Document.js

```diff
@@ -338,6 +338,13 @@
     'hasFeature',
   ]);
 
+  var originalCreateTreeWalker = document.createTreeWalker;
+  function createTreeWalker(root, whatToShow, filter, entityReferenceExpansion){
+    return originalCreateTreeWalker.call(document,unwrap(root), whatToShow, filter, entityReferenceExpansion);
+  }
+
+  document.createTreeWalker = createTreeWalker; // override
```

Comment:
hey, this is not quite the right way to do this, but it's close :)

Document is really subtle, because `window.document` cannot be overridden, so we put methods on both the Document wrapper and then a forwarder on the global document instance.

Could you move your implementation into the `mixin(Document.prototype, { ... })` block above in this file, then add this method name under `forwardMethodsToWrapper`

And perhaps add a test case along these lines:
```
// should work
wrap(document).createTreeWalker( ... )
// this too
document.createTreeWalker(...)
```

Wrapped documents need to have the `createTreeWalker` method, because they can be returned in various other ways (like `ownerDocument`)